qa/integration-tests/src/test/java/io/zeebe/broker/it/health/DiskSpaceRecoveryITTest.java

```diff
@@ -0,0 +1,138 @@
+
+public class DiskSpaceRecoveryITTest {
+  static ZeebeBrokerContainer zeebeBroker;
+  private static final Logger LOG = LoggerFactory.getLogger("TEST");
+  private static final String VOLUME_NAME = "tmpdata1";
+  private static String containerIPAddress;
+  private static Integer apiPort;
+  private static ElasticsearchContainer elastic;
+  private ZeebeClient client;
+
+  @BeforeClass
+  public static void setUpClass() {
+    final var elasticHostInternal = "http://elastic:9200";
+    zeebeBroker = createZeebe(elasticHostInternal);
+    final var network = zeebeBroker.getNetwork();
+    elastic = createElastic(network);
+    zeebeBroker.start();
+
+    apiPort = zeebeBroker.getMappedPort(26500);
+    containerIPAddress = zeebeBroker.getContainerIpAddress();
+  }
+
+  private static ZeebeBrokerContainer createZeebe(final String elasticHost) {
+    zeebeBroker =
+        new ZeebeBrokerContainer("current-test")
+            .withEnv(
+                "ZEEBE_BROKER_EXPORTERS_ELASTICSEARCH_CLASSNAME",
+                "io.zeebe.exporter.ElasticsearchExporter")
+            .withEnv("ZEEBE_BROKER_EXPORTERS_ELASTICSEARCH_ARGS_URL", elasticHost)
+            .withEnv("ZEEBE_BROKER_EXPORTERS_ELASTICSEARCH_ARGS_BULK_DELAY", "1")
+            .withEnv("ZEEBE_BROKER_EXPORTERS_ELASTICSEARCH_ARGS_BULK_SIZE", "1")
+            .withEnv("ZEEBE_BROKER_DATA_SNAPSHOTPERIOD", "1m")
+            .withEnv("ZEEBE_BROKER_DATA_LOGSEGMENTSIZE", "4MB")
+            .withEnv("ZEEBE_BROKER_DATA_DISKUSAGECOMMANDWATERMARK", "0.5")
+            .withEnv("ZEEBE_BROKER_DATA_LOGINDEXDENSITY", "1")
+            .withEnv("ZEEBE_BROKER_EXPORTERS_ELASTICSEARCH_ARGS_INDEX_MESSAGE", "true");
+
+    zeebeBroker
+        .getDockerClient()
+        .createVolumeCmd()
+        .withDriver("local")
+        .withDriverOpts(Map.of("type", "tmpfs", "device", "tmpfs", "o", "size=32m"))
+        .withName(VOLUME_NAME)
+        .exec();
+    final Volume newVolume = new Volume("/usr/local/zeebe/data");
+    zeebeBroker.withCreateContainerCmdModifier(
+        cmd ->
+            cmd.withHostConfig(cmd.getHostConfig().withBinds(new Bind(VOLUME_NAME, newVolume)))
+                .withName("zeebe-test"));
+
+    return zeebeBroker;
+  }
+
+  private static ElasticsearchContainer createElastic(final Network network) {
+    final ElasticsearchContainer container =
+        new ElasticsearchContainer(
+            "docker.elastic.co/elasticsearch/elasticsearch:"
+                + RestClient.class.getPackage().getImplementationVersion());
+
+    container.withNetwork(network).withEnv("discovery.type", "single-node");
+    container.withCreateContainerCmdModifier(cmd -> cmd.withName("elastic"));
+    return container;
+  }
+
+  @AfterClass
+  public static void tearDownClass() {
+    LoggerFactory.getLogger("Test").info(zeebeBroker.getLogs());
+    zeebeBroker.stop();
+    zeebeBroker.getDockerClient().removeVolumeCmd(VOLUME_NAME).exec();
```

Comment:
Not sure if we have to do this - `stop` from Testcontainers is actually more of a kill command, so it may also already be cleaning up the volumes, though we'd have to verify that.

Reply:
Volume should be closed explicitly. I just verified it.